arviz/tests/test_rcparams.py

```diff
@@ -135,6 +137,42 @@ def test_make_validate_choice(args, allow_none, typeof):
         assert value in accepted_values or value is None
 
 
+@pytest.mark.parametrize("allow_none", (True, False))
+@pytest.mark.parametrize("allow_auto", (True, False))
+@pytest.mark.parametrize("length", (2, 3))
+@pytest.mark.parametrize(
+    "args",
+    [
+        (False, (1, 2)),
+        ("Only ordered iterable", set(["a", "b", "c"])),
+        (False, None),
+        (False, "auto"),
+    ],
+)
```

Comment:
Given the complexity of the validator and how many cases it can take into account, I think it would be more clear to divide all these tests into several smaller test functions. For example: 
* one to test allow_none and allow_auto (testing with same parametrize True, False for both args and against a tuple, auto, None and a tuple-like string); 
* a second one to test the length (fixed a length vs None and against a tuple proper length, and tuple + tuple-like string of wrong lengths)
* a third one to test wrong inputs like set, non tuple-like string, a scalar...